torch_geometric/explain/algorithm/gnn_explainer.py

```diff
@@ -0,0 +1,539 @@
+        target_idx: Optional[int] = None,
+        node_index: Optional[int] = None,
+    ):
+        if target_idx is not None:
+            y_hat = y_hat[..., target_idx].unsqueeze(-1)
+            y = y[..., target_idx].unsqueeze(-1)
+
+        if node_index is not None:
+            loss_ = torch.cdist(y_hat[node_index], y[node_index])
+        else:
+            loss_ = torch.cdist(y_hat, y)
+
+        return loss_
+
+    def _loss_classification(
+        self,
+        y_hat: torch.Tensor,
+        y: torch.Tensor,
+        return_type: ModelReturnType,
+        target_idx: Optional[int] = None,
+        node_index: Optional[int] = None,
+    ):
+        if target_idx is not None:
+            y_hat = y_hat[target_idx]
+            y = y[target_idx]
+
+        y_hat = self._to_log_prob(y_hat, return_type)
+
+        if node_index is not None:
+            loss = -y_hat[node_index, y[node_index]]
+        else:
+            loss = -y_hat[0, y[0]]
+        return loss
+
+    def loss(
+        self,
+        y_hat: torch.Tensor,
+        y: torch.Tensor,
+        edge_mask_type: MaskType,
+        return_type: ModelReturnType,
+        node_index: Optional[int] = None,
+        target_idx: Optional[int] = None,
+        model_mode: ModelMode = ModelMode.regression,
+    ) -> torch.Tensor:
+
+        if model_mode == ModelMode.regression:
+            loss = self._loss_regression(y_hat, y, target_idx, node_index)
+        else:
+            loss = self._loss_classification(y_hat, y, return_type, target_idx,
+                                             node_index)
+
+        if edge_mask_type is not None:
+            m = self.edge_mask.sigmoid()
+            edge_reduce = getattr(torch, self.coeffs['edge_reduction'])
+            loss = loss + self.coeffs['edge_size'] * edge_reduce(m)
+            ent = -m * torch.log(m + self.coeffs['EPS']) - (
+                1 - m) * torch.log(1 - m + self.coeffs['EPS'])
+            loss = loss + self.coeffs['edge_ent'] * ent.mean()
+
+        m = self.node_mask.sigmoid()
+        node_feat_reduce = getattr(torch, self.coeffs['node_feat_reduction'])
+        loss = loss + self.coeffs['node_feat_size'] * node_feat_reduce(m)
+        ent = -m * torch.log(m + self.coeffs['EPS']) - (
+            1 - m) * torch.log(1 - m + self.coeffs['EPS'])
+        loss = loss + self.coeffs['node_feat_ent'] * ent.mean()
+
+        return loss
+
+    def supports(
+        self,
+        explainer_config: ExplainerConfig,
+        model_config: ModelConfig,
+    ) -> bool:
+        if model_config.task_level not in [
+                ModelTaskLevel.node, ModelTaskLevel.graph
+        ]:
+            logging.error("Model task level not supported.")
+            return False
+        if explainer_config.edge_mask_type not in [MaskType.object, None]:
+            logging.error("Edge mask type not supported.")
+            return False
+
+        if explainer_config.node_mask_type in [MaskType.both, None]:
```

Comment:
I would replace this with not in [...] to specifically clarify what is allowed. This would also make it more robust in case additional `node_mask_types` are added later.